pkg/virt-handler/virt-chroot/virt-chroot.go

```diff
@@ -20,28 +20,58 @@
 package virt_chroot
 
 import (
+	"fmt"
+	"os"
 	"os/exec"
 	"strings"
 
+	ps "github.com/mitchellh/go-ps"
+
 	"kubevirt.io/kubevirt/pkg/safepath"
 	"kubevirt.io/kubevirt/pkg/unsafepath"
 )
 
 const (
 	binaryPath     = "/usr/bin/virt-chroot"
-	mountNamespace = "/proc/1/ns/mnt"
+	MountNamespace = "/proc/1/ns/mnt"
 )
 
+// getParentProcessByName - return the parent process
+// this is needed to handle the case kubernetes and kubevirt
+// running inside a container and the mount namespace is different
+// from the host/node.
+func getParentProcess() (ps.Process, error) {
+	pid := os.Getpid()
+	process, err := ps.FindProcess(pid)
+	if err != nil {
+		return nil, err
+	}
+
+	process, err = ps.FindProcess(process.PPid())
+	if err != nil {
+		return nil, err
+	}
+	return process, nil
+}
+
+func GetHostNamespacePath() string {
+	process, err := getParentProcess()
+	if err == nil {
+		return fmt.Sprintf("/proc/%d/ns/mnt", process.Pid())
+	}
+	return MountNamespace
+}
+
 func getBaseArgs() []string {
-	return []string{"--mount", mountNamespace}
+	return []string{"--mount", GetHostNamespacePath()}
 }
 
 func GetChrootBinaryPath() string {
 	return binaryPath
 }
 
 func GetChrootMountNamespace() string {
-	return mountNamespace
+	return GetHostNamespacePath()
```

Comment:
`GetChrootMountNamespace() ` is also used by the tests reporter:

https://github.com/kubevirt/kubevirt/blob/4b41d9c7282452b9e4147463a6e68cae8849f45c/tests/reporter/kubernetes.go#L289-L300

I think with the new approach it will not work as expected. Therefore, I would directly use `/proc/1/ns/mnt` for the test suite.